lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -1613,6 +1613,104 @@ export class BrowserCacheManager extends CacheManager {
         this.setInteractionInProgress(false);
     }
 
+    /**
+     * Create request retry key to cache retry status
+     * @param correlationId
+     */
+    generateRequestRetriedKey(correlationId: string): string {
+        return `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.REQUEST_RETRY}.${correlationId}`;
+    }
+
+    /**
+     * Gets the request retry value from the cache
+     * @param correlationId
+     */
+    getRequestRetried(correlationId: string): number | null {
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        const cachedRetryNumber = this.getTemporaryCache(requestRetriedKey);
+        if (!cachedRetryNumber) {
+            return null;
+        }
+        return parseInt(cachedRetryNumber);
+    }
+
+    /**
+     * Sets the request retry value to "retried" in the cache
+     * @param correlationId
+     */
+    setRequestRetried(correlationId: string): void {
+        this.logger.trace("BrowserCacheManager.setRequestRetried called");
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        this.setTemporaryCache(requestRetriedKey, "1", false);
+    }
+
+    /**
+     * Removes all request retry values in the cache
+     */
+    removeRequestRetried(): void {
+        this.temporaryCacheStorage.getKeys().forEach((key) => {
```

Comment:
Looping through every key can be very expensive if sessionStorage is heavily utilized. I would suggest caching it with clientId instead of correlationId so we can do a direct lookup. I'm ok if you want to make that a follow up item though to get this out given this is temporary storage and won't need migration code.